html_css/v2/main-course/transforms-animation.md

```diff
@@ -0,0 +1,176 @@
+### Introduction
+
+Ready to add some animated flair to your website? CSS transitions and animations can add that "panache" to your elements so your websites will set you apart from the rest of the boring crowd. Animate your buttons, nav-bars, loading-bars, anything! Your imagination is your limit! Now let's dive in.  
+
+### Learning Outcomes
+
+* When should you use a CSS transition?
+* When should you use a CSS animation?
+* What are the long and short hand notations for both?
+* What should you keep in mind regarding repaints?
+* What are the two distinct parts of creating a keyframe animation?
+
+### Transitions
+CSS transitions let you animate a change from an element's initial state to an end state. Think of an ordinary button element with a white background. When your mouse is away from the button, it's just sitting there. Boring. Then when you hover your mouse cursor over the button the background color smoothly transitions from white, to grey, to black, over a period of time. This is a CSS transition.
+
+When your mouse cursor was away from the button, the button was in the initial state. When you hovered over it you introduced the end state, the hover state, while the transition happened as the color smoothly faded from white to black.
+
+In the above example we animated the CSS `background-color` property. While there are plenty of CSS properties you can animate, you can't animate them all. So keep [this list](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animated_properties) of "CSS animatable" properties handy so you can check whether or not you can do the transition you want.
+
+Now let's make a basic transition. There are two ways of writing a transition. One is the long form and the other is the short form. The long form looks like this:
```

Comment:
People should know about shorthand properties already so I'd just use that term instead of the phrase 'two ways'... We're already referring to `transitions` and `keyframe animations` as "two ways" and I don't want to confuse that.

Instead of 'long form' i would probably say something like:

> The `transition` property is a shorthand for the following four properties: